### DISCUSSION TITLE
Route53 Create a hosted zone using a delegationsetId .NET SDK

### DISCUSSION THREAD

**Discussion by mvestuto** [aws/aws-sdk-net#3170] [Q&A] [answered]
Howdy, I'm using the .NET SDK 3.7.300.XX.
I need to create hosted zones using a delegationset. We have a number of existing domains that use this delegation set, and any new domains that I create need to also be created this delgation set.

From what I understand, the delegationset can't be changed after the zone has been created.
When creating a createHostedZoneRequest, there is no property to define the delegationsetId.

Is it possible to create a zone with delegation, or define a delegationset on an existing domain with the .net sdk?

**Comment by dscpinheiro** [comment, marked as answer]
In version `3.7.302.13` of the `AWSSDK.Route53` package, I see a `DelegationSetId` property in `CreateHostedZoneRequest`, is that what you were looking for?
```csharp
using Amazon.Route53;
using Amazon.Route53.Model;

var route53 = new AmazonRoute53Client();
await route53.CreateHostedZoneAsync(new CreateHostedZoneRequest
{
    DelegationSetId = ""
});
```

From our [docs](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/Route53/TCreateHostedZoneRequest.html):
```
If you want to use the same name servers for multiple public hosted zones, you can optionally associate a reusable delegation set with the hosted zone. See the DelegationSetId element. 
```

**Reply by mvestuto** [reply]
That is what I was looking for. I was unaware of the Async method.
I also had a different issue where where this method was not available at all. I had an older reference of AWSSDK which required an alias. After aliasing I can now get to the method. Thanks.
